src/test/java/uk/gov/ons/ctp/integration/envoycuc/mockclient/MockClient.java

```diff
@@ -0,0 +1,292 @@
+package uk.gov.ons.ctp.integration.envoycuc.mockclient;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.PostConstruct;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang3.time.DateUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.ons.ctp.common.domain.CaseType;
+import uk.gov.ons.ctp.common.domain.UniquePropertyReferenceNumber;
+import uk.gov.ons.ctp.integration.common.product.model.Product;
+import uk.gov.ons.ctp.integration.envoycuc.client.RateLimiterClientRequest;
+import uk.gov.ons.ctp.integration.envoycuc.client.TestClient;
+import uk.gov.ons.ctp.integration.ratelimiter.client.RateLimiterClient;
+import uk.gov.ons.ctp.integration.ratelimiter.model.CurrentLimit;
+import uk.gov.ons.ctp.integration.ratelimiter.model.LimitStatus;
+import uk.gov.ons.ctp.integration.ratelimiter.model.RateLimitResponse;
+
+@Data
+@NoArgsConstructor
+@Component
+public class MockClient implements TestClient {
```

Comment:
I think this is a very useful bit of code, and worth ensuring its correct.

I know this would be testing the test framework, but I would consider some unit tests for this, since I personally wouldn't be 100% confident it works as expected, even despite all those cucumber scenarios going through it !